tests/next_tests/unit_tests/iterator_tests/transforms_tests/test_cse.py

```diff
@@ -177,3 +182,67 @@ def test_if_can_deref_eligible_extraction():
 
     actual = CSE().visit(testee)
     assert actual == expected
+
+
+def test_extract_subexpression_conversion_to_assignment_stmt_form():
```

Comment:
Why this (not sure how to call it) test instead of just a unittests?

Reply:
It's also doesn't have full coverage because it's neglecting the 3 third tuple element.

Reply:
TODO: add comment as discussed